File: Src/Workspaces/VisualBasic/Extensions/CommonSyntaxTokenExtensions.vb
Imports System.Runtime.CompilerServices
Imports Microsoft.CodeAnalysis.Common

Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
    Module CommonSyntaxTokenExtensions
#If REMOVE Then
        <Extension()>
        Public Function IsKindOrHasMatchingText(token As SyntaxToken, kind As SyntaxKind) As Boolean
            Return CType(token, SyntaxToken).IsKindOrHasMatchingText(kind)
        End Function

        <Extension()>
        Public Function HasMatchingText(token As SyntaxToken, kind As SyntaxKind) As Boolean
            Return CType(token, SyntaxToken).HasMatchingText(kind)
        End Function

        <Extension()>
        Public Function IsParentKind(token As SyntaxToken, kind As SyntaxKind) As Boolean
            Return CType(token, SyntaxToken).IsParentKind(kind)
        End Function

        <Extension()>
        Public Function MatchesKind(token As SyntaxToken, kind As SyntaxKind) As Boolean
            Return CType(token, SyntaxToken).MatchesKind(kind)
        End Function

        <Extension()>
        Public Function MatchesKind(token As SyntaxToken, kind1 As SyntaxKind, kind2 As SyntaxKind) As Boolean
            Return CType(token, SyntaxToken).MatchesKind(kind1, kind2)
        End Function

        <Extension()>
        Public Function MatchesKind(token As SyntaxToken, ParamArray kinds As SyntaxKind()) As Boolean
            Return CType(token, SyntaxToken).MatchesKind(kinds)
        End Function
#End If
    End Module
End Namespace

File: Src/Workspaces/CSharp/Extensions/CommonSyntaxTokenExtensions.cs
using Microsoft.CodeAnalysis.Common;

namespace Microsoft.CodeAnalysis.CSharp.Extensions
{
    internal static class CommonSyntaxTokenExtensions
    {
#if REMOVE
        public static bool IsKindOrHasMatchingText(this SyntaxToken token, SyntaxKind kind)
        {
            return ((SyntaxToken)token).IsKindOrHasMatchingText(kind);
        }

        public static bool HasMatchingText(this SyntaxToken token, SyntaxKind kind)
        {
            return ((SyntaxToken)token).HasMatchingText(kind);
        }

        public static bool IsParentKind(this SyntaxToken token, SyntaxKind kind)
        {
            return ((SyntaxToken)token).IsParentKind(kind);
        }

        public static bool MatchesKind(this SyntaxToken token, SyntaxKind kind)
        {
            return ((SyntaxToken)token).MatchesKind(kind);
        }

        public static bool MatchesKind(this SyntaxToken token, SyntaxKind kind1, SyntaxKind kind2)
        {
            return ((SyntaxToken)token).MatchesKind(kind1, kind2);
        }

        public static bool MatchesKind(this SyntaxToken token, params SyntaxKind[] kinds)
        {
            return ((SyntaxToken)token).MatchesKind(kinds);
        }
#endif
    }
}
